### ISSUE TITLE
Minecraft Java Edition support

### ISSUE THREAD

**Issue by chrrybmb64** [Heroic-Games-Launcher/HeroicGamesLauncher#4039]
### Feature description

I think Minecraft Java Edition support would be a good addition to Heroic since it's not available on any game store, meaning the only way to download Minecraft is via it's own launcher. Minecraft support in Heroic would remove the need for that launcher, and also help centralize non-steam games into one launcher.

### Alternatives

Maybe [Prism Launcher](https://github.com/PrismLauncher/PrismLauncher) can be integrated into Heroic in some way? Since it's open source and has a ton of features perhaps doing that would be better than starting from scratch.

**Comment by MattSturgeon** [comment]
This is an interesting idea.

Personally, I'd need a _lot_ of Minecraft-specific features to convince me to move from prism to heroic.

For instance, heroic currently isn't very good at having multiple instances of a game (something that is very common when playing Minecraft); you can copy-paste the game install and "add" the copy to heroic, but it will no longer be associated with the game store it was originally downloaded from.

I also often use prism's ability to quickly download a few mods from modrinth or curseforge.

There's also skin selection/uploading. And login into multiple accounts.

These are all reasons why someone may prefer a more feature packed Minecraft launcher, but none of them block basic support. I just wonder whether it is worth the effort to support Minecraft when many users will continue to use dedicated launchers?

At the bare minimum, you would need to implement Microsoft/XBOX oauth, fetching game version metadata from mojang's REST API and downloading the game, it's listed dependencies, and a JDK.

**Comment by chrrybmb64** [comment]
> Personally, I'd need a lot of Minecraft-specific features to convince me to move from prism to heroic.

I agree. This is why I think it's a good idea to explore whether Prism's code can be integrated into Heroic. That would save a lot of time that would be spent trying to catch up to Prism's long feature list.

> At the bare minimum, you would need to implement Microsoft/XBOX oauth, fetching game version metadata from mojang's REST API and downloading the game, it's listed dependencies, and a JDK.

Some users may already have a JDK or JRE installed, so detecting a pre-installed one and using it to launch Minecraft should be implemented as well.

**Comment by Twig6943** [comment]
I mean this is kinda pointless since there are already feature rich mc launcher out there such as prism

**Comment by arielj** [comment]
I think this is really out of scope of Heroic given there's better alternatives. It would be too much work for such a small use case

**Comment by MattSturgeon** [comment]
> I think this is really out of scope

Note github has options for [closing issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/administering-issues/closing-an-issue) in other ways; "completed" isn't really the best option for an issue deemed out of scope, that's more like "invalid" or "won't fix" which shows up as _grey_ instead of _purple_ for easy distinction.
